src/libraries/System.Private.Xml/src/System/Xml/Serialization/CodeGenerator.cs

```diff
@@ -53,6 +53,10 @@ internal static bool IsNullableGenericType(Type type)
             return type.Name == "Nullable`1";
         }
 
+#if DEBUG
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2070:UnrecognizedReflectionPattern",
+            Justification = "Debug only code, we don't ship debug binaries.")]
+#endif
         internal static void AssertHasInterface(Type type, Type iType)
```

Comment:
Nit: Can we put `[Conditional("DEBUG")]` on the method itself to eliminate all the ifdeffing?